compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -652,7 +652,13 @@ impl<'tcx> Collector<'tcx> {
     ) -> DllImport {
         let span = self.tcx.def_span(item);
 
-        // this logic is similar to `Target::adjust_abi` (in rustc_target/src/spec/mod.rs) but errors on unsupported inputs
+        // This `extern` block should have been checked for general ABI support before, but let's
+        // double-check that.
+        assert!(self.tcx.sess.target.is_abi_supported(abi));
+
+        // This logic is similar to `Target::adjust_abi` (in rustc_target/src/spec/mod.rs) but we
+        // need more detail than those adjustments, and we can't support all ABIs that are generally
+        // supported.
```

Comment:
...I... believe this comment might need to be rephrased, and I'm not sure how exactly.

Reply:
I've given it a shot.